### PR TITLE
fix: 修复序号对齐问题， close #1412

### LINKED ISSUE #1412: 🐛 序号没有垂直居中
### 🏷 Version

| Package      | Version |
| -------------- | --------- |
| @antv/s2       |     1.19.0     |
| @antv/s2-react |     1.17.0     |
| @antv/s2-vue   |     1.0.1    |

### Sheet Type

- [x] PivotSheet
- [ ] TableSheet
- [ ] GridAnalysisSheet
- [ ] StrategySheet

### 🖋 Description
image
image

### PR DESCRIPTION
**Pull request by wjgogogo**
### 👀 PR includes

✨ Feature

- [ ] New feature

🎨 Enhance

- [x] 增加针对序号文字的主题配置项 
- [ ] Refactoring
- [ ] Change the UI
- [ ] Improve the performance
- [ ] Type optimization

🐛 Bugfix
关联PR: #1364 
- [x] Solve the issue and close #1412 

🔧 Chore

- [ ] Test case
- [ ] Docs / demos update
- [ ] CI / workflow
- [ ] Release version
- [ ] Other ()

### 🖼️ Screenshot

| Before | After |
| ------ | ----- |
| imageimage    |  imageimage     |

### 🔍 Self Check before Merge

- [ ] Add or update relevant Docs.
- [ ] Add or update relevant Demos.
- [ ] Add or update relevant TypeScript definitions.

### CODE REVIEW

File: packages/s2-core/src/cell/row-cell.ts [hunk 1]
```diff
@@ -364,16 +364,16 @@ export class RowCell extends HeaderCell {
...
-      return bolderText;
+      style = measureText || text;
+    } else if (this.isBolderText()) {
+      style = bolderText;
+    } else {
+      style = text;
     }
 
-    return text;
+    return { ...style, textBaseline: 'top' };
```
Comment (wjgogogo) [inline packages/s2-core/src/cell/row-cell.ts:376]: 这个地方必须保持是`top`，因为下面做滚动计算时，是按照`top`的方式计算坐标
https://github.com/antvis/S2/blob/2ca8048457d19ed54aa8b05fe5fcb89e46787eba/packages/s2-core/src/cell/row-cell.ts#L447-L453
如果不写死，就会出现：使用`top`的方式计算坐标，然后按照用户配置的`textBaseline`渲染：

File: packages/s2-core/src/facet/header/series-number.ts [hunk 2]
```diff
@@ -157,29 +157,26 @@ export class SeriesNumberHeader extends BaseHeader<BaseHeaderConfig> {
...
-      x,
-      y,
-      width: cellWidth,
-      height: cellHeight,
-      isLeaf,
-      isTotals,
-    } = cellData;
+    const textStyle = {
+      ...this.getStye().seriesText,
+      textBaseline: 'top' as const,
```
Comment (wjgogogo) [inline packages/s2-core/src/facet/header/series-number.ts:176]: 原因同上

File: packages/s2-core/src/facet/header/series-number.ts [hunk 3]
```diff
@@ -157,29 +157,26 @@ export class SeriesNumberHeader extends BaseHeader<BaseHeaderConfig> {
...
-    const cellTheme = this.headerConfig.spreadsheet.theme.rowCell.cell;
+    const cellTheme = this.getStye().cell;
 
     const { position: horizontalPosition, style: horizontalStyle } =
       getBorderPositionAndStyle(CellBorderPosition.BOTTOM, cellData, cellTheme);
 
     renderLine(group as Group, horizontalPosition, horizontalStyle);
   }
 
+  private getStye() {
```
Comment (lijinke666) [inline packages/s2-core/src/facet/header/series-number.ts:168]: getStyle ?

**Review by lijinke666 — APPROVED** [review]

**Review by lijinke666 — APPROVED** [review]

**Pull request merged**